api/src/index.ts

```diff
@@ -0,0 +1,204 @@
+import { GearApi, getWasmMetadata, Hex, Metadata, PayloadType } from '@gear-js/api';
+import { UploadProgramModel, UploadProgram, ProgramState, MessageModel, Action, Account } from "gear-program-interface-core";
+import { payloads } from './payloads';
+import * as Blake2b from 'blake2b';
+
+export enum Move {
+  ROCK,
+  PAPER,
+  SCISSORS,
+  LIZARD,
+  SPOCK,
+}
+
+async function getMeta() {
+  const metaPath = '../contract_files/rock_paper_scissors.meta.wasm';
+  return Buffer.from(await (await fetch(metaPath)).arrayBuffer());
+}
+
+async function getCodeFile() {
+  const path = '../contract_files/rock_paper_scissors.opt.wasm';
+  return new File([await (await fetch(path)).blob()], path);
+}
+
+export async function deploy(
+  gearApi: GearApi,
+  account: Account,
+  bet_size: number,
+  players: [Hex],
+  programIdHandler: (id: Hex) => void,
+) {
+  const metaBuffer = await getMeta();
+  const code = await getCodeFile();
+  const meta = await getWasmMetadata(metaBuffer);
+  const initPayload = payloads.init(bet_size, players);
+
+  const programOptions: UploadProgramModel = {
+    meta,
+    value: 0,
+    initPayload,
+  };
+
+  UploadProgram(gearApi, account, code, programOptions, programIdHandler)
+}
+
+async function sendAction(
+  gearApi: GearApi,
+  programId: Hex,
+  account: Account,
+  payload: PayloadType,
+  eventHandler: (event: any) => void,
+) {
+  const metaBuffer = await getMeta();
+  const meta = (metaBuffer !== null && metaBuffer !== undefined) ? await getWasmMetadata(metaBuffer) : null;
```

Comment:
```suggestion
  const meta = metaBuffer ? await getWasmMetadata(metaBuffer) : null;
```

Reply:
And actually, how will you send message without meta? Maybe it is better to throw an error here in case the metaBuffer doesn't exist